infrastructure/maintenance-container/Dockerfile

```diff
@@ -2,14 +2,14 @@
 FROM node:18.16.0 as node
 
 # base
-FROM ubuntu:22.04
+FROM ubuntu:23.10
```

Comment:
For consistency purposes, I think all the workflows with set Ubuntu versions ought to be updated too

Reply:
I don't think 23.10 is available for Github workflows.  The latest is 22.04.  They probably just stick to the LTS releases.  
https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

It's not as big a deal since we use the shivammathur/setup-php@v2 action to specify the version of PHP we want to run in the workflow.

Reply:
Ah okay, something to come back to later then